test/services/DateServiceSpec.scala

```diff
@@ -38,35 +43,126 @@ class DateServiceSpec extends TestSupport with FeatureSwitching {
     Some("1234567890"), Some("12345-credId"), Some(Individual), None
   )(FakeRequest())
 
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(appConfig)
+  }
+
+  override def afterEach(): Unit = {
+    super.beforeEach()
+    reset(appConfig)
+  }
+
+  def setupGetCurrentTestMocks(timeMachineFS: Boolean = false, timeMachineAddYearFS: Boolean = false,
+                               addYears: Int = 0, timeMachineSetDateFS: Boolean = false, setDay: Int = 0, setMonth: Int = 0): Unit = {
+    when(appConfig.isTimeMachineEnabled).thenReturn(timeMachineFS)
+    when(appConfig.isTimeMachineAddYearEnabled).thenReturn(timeMachineAddYearFS)
+    when(appConfig.timeMachineAddYears).thenReturn(addYears)
+    when(appConfig.isTimeMachineSetDateEnabled).thenReturn(timeMachineSetDateFS)
+    when(appConfig.timeMachineSetDateDay).thenReturn(setDay)
+    when(appConfig.timeMachineSetDateMonth).thenReturn(setMonth)
+  }
+
   def fixture(date: String) = new {
-    val mockedTestDateService = new DateService() {
+    val mockedTestDateService: DateService = new DateService() {
       override def getCurrentDate: LocalDate = {
-//        val timeMachineIsOn: Boolean = isEnabled(TimeMachineAddYear)(testUser)
+        //        val timeMachineIsOn: Boolean = isEnabled(TimeMachineAddYear)(testUser)
 
-//        if (timeMachineIsOn) {
-//          LocalDate.parse(date).plusYears(1)
-//        } else {
-          LocalDate.parse(date)
-//        }
+        //        if (timeMachineIsOn) {
+        //          LocalDate.parse(date).plusYears(1)
+        //        } else {
+        LocalDate.parse(date)
```

Comment:
as its been mention quite a while ago, this can fail for some dates on the OS where dates got a different format say: YYYYMMDD => YYYYDDMM, so success of this operation is not guarantied